jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/profiles/ProfileCredentialProviderFactory.kt

```diff
@@ -80,8 +78,10 @@ class ProfileCredentialProviderFactory : CredentialProviderFactory, Disposable {
         // Load the initial data, then start the background watcher
         loadProfiles(credentialLoadCallback, true)
 
-        profileWatcher.start(onFileChange = {
-            loadProfiles(credentialLoadCallback, false)
+        ProfileWatcher.getInstance().addListener(object : ProfileWatcher.Listener {
+            override fun profilesUpdated() {
```

Comment:
SAM conversion doesnt seem to work if its Kotlin defined (1.4 will support it https://blog.jetbrains.com/kotlin/2020/03/kotlin-1-4-m1-released/)... 

I avoided making in-place Lambda callback in case we ever wanted to extend this interface...but maybe that was the wrong call.

Reply:
Since this is tied to our code and should probably not be considered a public API, I think I can go with Kotlin lambda callback. I'll change